### DISCUSSION TITLE
Insert signature in a page

### DISCUSSION THREAD

**Discussion by AlexanderQuinaluiza** [lsnepomuceno/laravel-a1-pdf-sign#74] [Q&A] [answered]
I try to enter the stamp of the digital signature on a defined page using the following code: 
```

$resource = $pdf->setImage($imagePath, 10,10, 20,20, 3) 
                    ->setFileName('new-file-name.pdf') 
                    ->setInfo( 
                         $name,
                         $location,
                         $reason,
                         $contactInfo 
                    )
                    ->setHasSignedSuffix(false)
                    ->signature();
```

But anyway the signature is entered on the last page, any idea how to solve this?

**Comment by lsnepomuceno** [comment]
Hello @AlexanderQuinaluiza ,

So that I can help you better, please let me know which version of the package you are using.

I'm starting the validations in the latest version.

**Reply by AlexanderQuinaluiza** [reply]
The version "lsnepomuceno/laravel-a1-pdf-sign": "^1.0"

**Comment by lsnepomuceno** [comment, marked as answer]
Hello @AlexanderQuinaluiza ,

The fix was implemented in PR https://github.com/lsnepomuceno/laravel-a1-pdf-sign/pull/85 and is now available from version [v1.0.7](https://github.com/lsnepomuceno/laravel-a1-pdf-sign/releases/tag/1.0.7) 

I kindly ask you to mark this answer as correct so that it will be indexed and help other developers.

**Comment by gmanducaPilar** [comment]
Hello Lucas,
I'm trying to insert the certificate .pfx in the pdf file, but The system show me an error. I think that I'm creating wrong the certificate.
I tried exporting with WIndows => internet option => certificates => exports ..
another way I tried to create with Open SSL, but the openssl create the certificate with PKCS12 not with PKCS7, so the system show me

"The file is unsigned or the signature is not compatible with the PKCS7 type. "

Do you know how I can create the certificate compatible with PKCS7, 
thank you
Gastón

**Comment by lsnepomuceno** [comment]
Hello @gmanducaPilar ,

Interesting, I had not yet seen this type of need within the package, but as it is a real case, I can evaluate the possibility of including this type of certificate in the signature of PDF files.

Is it possible to send me the certificate by email?

I will validate how I convert to the current format of the package, if it works, I will already include the functionality in the next version.

Follow my email: lsn.nepomuceno@gmail.com

**Reply by gmanducaPilar** [reply]
Hello Lucas , I sent the certificate by e-mail, do you have some time to check ot ? or could you tell me how I can create a certificate to work with the system. 

Best regards

**Reply by lsnepomuceno** [reply]
@gmanducaPilar ,

I received the certificate and started the validations.

The structure is different than what the package supports, so I'm validating the type conversion for the document signature without breaking the certificate data.

I'll get back to you as soon as possible with a new version already supporting your certificate.
